### ISSUE TITLE
rust_doc_test requires a local C++ compiler

### ISSUE THREAD

**Issue by sayrer** [bazelbuild/rules_rust#845]
I ran into problems with running this locally, on a VM with no C++ toolchain available.

```
---- path/to/lib.rs - DemoStruct::demo_method (line 131) stdout ----
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)
```

**Comment by UebelAndre** [comment]
I think this is expected? `rustc` requires some information from the toolchain so since we're building a test here, we'd need that toolchain to run the compiler, right?

**Comment by sayrer** [comment]
I don't think it is expected. The other rules, `rust_test`, `cargo_build_script`, etc all work without a local C++ toolchain. Maybe something similar to #664 is needed.

**Comment by wt** [comment]
Can you please try this with Bazel 5.0 and see if this worksw with the C++ autoconfigured toolchain?
